src/include/ev.h

```diff
@@ -0,0 +1,396 @@
+   int (*signal_start)(struct ev_loop* loop, struct ev_signal* watcher); /**< Starts a signal watcher in the event loop. */
+   int (*signal_stop)(struct ev_loop* loop, struct ev_signal* watcher); /**< Stops a signal watcher in the event loop. */
+   int (*periodic_init)(struct ev_loop* loop, struct ev_periodic* watcher); /**< Initializes a periodic watcher. */
+   int (*periodic_start)(struct ev_loop* loop, struct ev_periodic* watcher); /**< Starts a periodic watcher in the event loop. */
+   int (*periodic_stop)(struct ev_loop* loop, struct ev_periodic* watcher); /**< Stops a periodic watcher in the event loop. */
+};
+
+/**
+ * @struct ev_loop
+ * @brief Main event loop structure.
+ *
+ * Manages the event loop, including I/O, signal, and periodic watchers.
+ * It handles the execution and coordination of events using the specified backend.
+ *
+ * TODO maybe `struct ev_loop` could be separated into ev_loop{,_uring,_epoll,_kqueue} so that it
+ * could be dinamically plugged into ev_loop.
+ */
+struct ev_loop
+{
+   volatile bool running;               /**< Flag indicating if the event loop is running. */
+   atomic_bool atomic_running;          /**< Atomic flag for thread-safe running state. */
+   struct ev_context ctx;               /**< Context containing configuration and state. */
+
+   struct ev_io ihead;                  /**< Head of the I/O watchers linked list. */
+   struct ev_signal shead;              /**< Head of the signal watchers linked list. */
+   struct ev_periodic phead;            /**< Head of the signal watchers linked list. */
+
+   sigset_t sigset;                     /**< Signal set used for handling signals in the event loop. */
+
+   struct ev_ops ops;                   /**< Backend operations for the event loop. */
+
+   struct configuration* config;        /**< Pointer to pgagroal global configuration. */
+
+#if HAVE_URING
+   struct io_uring_cqe* cqe;
+
+   struct io_uring ring;
+   struct io_buf_ring in_br;
+   struct io_buf_ring out_br;
+   /* TODO: Consider removing the usage of .bid */
+   int bid;                             /**< io_uring: Next buffer id. */
+
+   /* TODO: Implement iovecs.
+    *   int iovecs_nr;
+    *   struct iovec *iovecs;
+    */
+
+#endif /* HAVE_URING */
+
+#if HAVE_EPOLL
+   int epollfd;                         /**< File descriptor for the epoll instance (used with epoll backend). */
+#endif
+
+#if HAVE_KQUEUE
+   int kqueuefd;                        /**< File descriptor for the kqueue instance (used with kqueue backend). */
+#endif
+
+   void* buffer;                        /**< Pointer to a buffer used to read in bytes. */
+
+};
+
+typedef void (*io_cb)(struct ev_loop*, struct ev_io* watcher, int err);
+typedef void (*signal_cb)(struct ev_loop*, struct ev_signal* watcher, int err);
+typedef void (*periodic_cb)(struct ev_loop*, struct ev_periodic* watcher, int err);
+
+/** This set of functions initializes, starts, breaks, and destroys an event loop.
+ * @param w:
```

Comment:
Documentation

Reply:
It is not even clear to what it is referring to with all the `@param`, wrong place?